tests/test_common_features.py

```diff
@@ -1,12 +1,177 @@
+import functools
+from importlib import import_module
 from pytest_dash.utils import (
     import_app,
     wait_for_element_by_id,
+    wait_for_text_to_equal,
+    wait_for_element_by_css_selector,
 )
+import dash
+from dash.dependencies import Input, Output, State
+import dash_html_components as html
+import dash_core_components as dcc
+
+COMPONENT_PYTHON_BASE = 'python'
+COMPONENT_REACT_BASE = 'react'
+
+PROP_TYPES = {
+    'int': int,
+    'float': float,
+    'bool': bool,
+    'str': str,
+    'array': lambda x: [float(el) for el in x.split(',')]
+}
 
 
 def access_demo_app(dash_threaded, selenium, app_name):
-    """mimic a user click on the app link from the gallery"""
+    """Mimic a user click on the app link from the gallery."""
     dash_bio_index = import_app('index')
     dash_threaded(dash_bio_index)
     link = wait_for_element_by_id(selenium, 'app-link-id-{}'.format(app_name))
     link.click()
+
+
+def init_demo_app(app_name):
+    # allow the decorator to take the app_name as an argument
+    def decorator_init_demo_app(func):
+        # preserves the __name__ and __doc__ values of the decorated func
+        @functools.wraps(func)
+        def wrapper_init_demo_app(dash_threaded, selenium):
+            # start the index.py app in a thread and move to the component URL
+            access_demo_app(dash_threaded, selenium, app_name)
+            # execute the test function from there
+            func(dash_threaded, selenium)
+        return wrapper_init_demo_app
+    return decorator_init_demo_app
+
+
+def import_component(component_name):
+    """Imports a component from the dash_bio package given its name.
+        :param component_name: (string) formatted like 'part_1-part_2-...-part_N', where each
+        part_i are in lower case.
+        :return: a dash_bio.ComponentName class, where ComponentName is the same as
+        component_name with '-' removed and each part_i having the first letter in upper case.
+    Example: 'needle-plot' will return dash_bio.NeedlePlot class.
+    """
+    name_parts = component_name.split('-')
+    component_name = ''
+    # loop over the parts
+    for name in name_parts:
+        component_name = '{}{}{}'.format(component_name, name[0].upper(), name[1:])
+    component_module = import_module('.{}'.format(component_name), package='dash_bio')
+    return getattr(component_module, component_name)
+
+
+def create_test_layout(app_name, component_base, **kwargs):
+    """Create a simple layout for an app to test component's props."""
+    component_id = 'test-{}-component'.format(app_name)
+
+    if component_base == COMPONENT_PYTHON_BASE:
+        component = dcc.Graph(id=component_id)
+    elif component_base == COMPONENT_REACT_BASE:
+        # equivalent to `component = dashbio.ComponentName(id=component_id)`
+        component_class = import_component(app_name)
+        component = component_class(id=component_id, **kwargs)
```

Comment:
`**kwargs` are useful here if a component needs to be initialized with some data for example